### PR TITLE
build: bring in a later compiler for Windows on Arm

### PR DESCRIPTION
**Pull request by richard-townsend-arm**
#### Description of Change

Due to a code-gen defect affecting virtual method calls in the official Clang used in the M76 branch, it's necessary to use a slightly newer compiler for Windows. [The issue has been fixed upstream](https://bugs.llvm.org/show_bug.cgi?id=41997), but for various reasons it was not possible to update the compiler to a version that worked on all devices in M76. This is only intended to be a temporary workaround and will be removed when upstream Chromium updates to a Clang version which fixes the problem.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Added experimental support for building for Windows on Arm

### CODE REVIEW

**Comment by richard-townsend-arm** [comment]
@jkleinsc @MarshallOfSound

File: docs/development/build-instructions-gn.md [hunk 1]
```diff
@@ -194,16 +194,39 @@ $ gn gen out/Debug-x86 --args='... target_cpu = "x86"'
...
 
 See the GN reference for allowable values of [`target_os`][target_os values]
-and [`target_cpu`][target_cpu values]
+and [`target_cpu`][target_cpu values].
 
 [target_os values]: https://gn.googlesource.com/gn/+/master/docs/reference.md#built_in-predefined-variables-target_os_the-desired-operating-system-for-the-build-possible-values
 [target_cpu values]: https://gn.googlesource.com/gn/+/master/docs/reference.md#built_in-predefined-variables-target_cpu_the-desired-cpu-architecture-for-the-build-possible-values
 
+#### Windows on Arm (experimental)
+To cross-compile for Windows on Arm, [follow Chromium's guide](https://chromium.googlesource.com/chromium/src/+/refs/heads/master/docs/windows_build_instructions.md#Visual-Studio) to get the necessary dependencies, SDK and libraries, then build with `ELECTRON_BUILDING_WOA=1` in your environment before running `gclient sync`.
```
Comment (felixrieseberg) [inline docs/development/build-instructions-gn.md]: Nit: Newline after title

```suggestion

To cross-compile for Windows on Arm, [follow Chromium's guide](https://chromium.googlesource.com/chromium/src/+/refs/heads/master/docs/windows_build_instructions.md#Visual-Studio) to get the necessary dependencies, SDK and libraries, then build with `ELECTRON_BUILDING_WOA=1` in your environment before running `gclient sync`.
```

**Comment by jkleinsc** [comment]
WOA build with this change here: https://ci.appveyor.com/project/electron-bot/electron-ldhmv/builds/25013835

**Comment by jkleinsc** [comment]
Merging as failing mac test is unrelated to PR and known to not work for fork PRs.

File: patches/common/chromium/woa_compiler_workaround.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
+From b3414d055399d0a21f6166a536467ea752b2aa8a Mon Sep 17 00:00:00 2001
```
Comment (MarshallOfSound) [inline patches/common/chromium/woa_compiler_workaround.patch]: Just FYI @jkleinsc @felixrieseberg @richard-townsend-arm this patch file is invalid.  Although it works it wasn't generated by the `export-patches` script which means that the next time someone runs import/export on their patch directory this file will change.  E.g. The `From` line here should be `000000000...` and the git version `2.19.1.windows.1` shouldn't be present.

`./electron/script/git-export-patches -o ./electron/patches/common/chromium` is the command to export a patch applied on top of chromium 👍

Comment (richard-townsend-arm) [reply patches/common/chromium/woa_compiler_workaround.patch]: Sounds like the kind of thing a presubmit hook should check for :D

**Pull request merged**